Sources/CasePaths/Optional+CasePathable.swift

```diff
@@ -86,3 +86,26 @@ extension Optional.AllCasePaths: Sequence {
     [\.none, \.some].makeIterator()
   }
 }
+
+extension Optional where Wrapped: CasePathable {
+  /// Tests the associated value of an optional case.
+  ///
+  /// ```swift
+  /// @CasePathable
+  /// enum UserAction {
+  ///   case home(HomeAction)
+  ///   case settings(SettingsAction)
+  /// }
+  ///
+  /// let userAction: UserAction? = .home(.onAppear)
+  /// userAction.is(\.home)      // true
+  /// userAction.is(\.settings)  // false
+  ///
+  /// let userActions: [UserAction] = [.home(.onAppear), .settings(.subscribeButtonTapped)]
+  /// userActions.filter { $0.is(\.home) }      // [UserAction.home(.onAppear)]
+  /// userActions.filter { $0.is(\.settings) }  // [UserAction.settings(.subscribeButtonTapped)]
+  /// ```
```

Comment:
Since `Optional` _is_ `CasePathable` I think we can hide this overload's primary use as an aid in autocomplete.

```suggestion
  @_disfavoredOverload
  @_documentation(visibility:internal)
```